src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -891,18 +884,15 @@ void unsubscribe(NatsSubscription sub, int after) {
     }
 
     void sendUnsub(NatsSubscription sub, int after) {
-        String sid = sub.getSID();
-        CharBuffer protocolBuilder = CharBuffer.allocate(this.options.getMaxControlLine());
-        protocolBuilder.append(OP_UNSUB);
-        protocolBuilder.append(" ");
-        protocolBuilder.append(sid);
+        // allocate the proto length + 19 + 10 (sid is a long, 19 bytes max, after is an int 10 bytes max)
+        ByteArrayBuilder bab = new ByteArrayBuilder(OP_UNSUB_PROTOCOL_LEN + 29)

```

Comment:
Nice...

Reply:
This doesn't allocate the correct buffer size which causes unnecessary allocations/copying.

Reply:
`OP_UNSUB_PROTOCOL_LEN + 29` is the maximum number of bytes needed as noted in the comment just above the line of code. The builder code actually allocates in chunks of 32 bytes, so 64 bytes is probably allocated here.

Reply:
Allocating too large a buffer just like too small causes performance issues as both result in extra buffer copying.